spring-cloud-task-stream/src/main/java/org/springframework/cloud/task/batch/listener/support/TaskEventProperties.java

```diff
@@ -0,0 +1,93 @@
+package org.springframework.cloud.task.batch.listener.support;
+
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.core.Ordered;
+
+/**
+ *  @author Ali Shahbour
+ */
+@ConfigurationProperties(prefix = "spring.cloud.task.events")
+public class TaskEventProperties {
+
+    private ListenerProperties jobExecution = new ListenerProperties(Ordered.HIGHEST_PRECEDENCE);
+    private ListenerProperties stepExecution = new ListenerProperties(Ordered.HIGHEST_PRECEDENCE);
+    private ListenerProperties itemRead = new ListenerProperties(Ordered.HIGHEST_PRECEDENCE);
+    private ListenerProperties itemProcess = new ListenerProperties(Ordered.HIGHEST_PRECEDENCE);
+    private ListenerProperties itemWrite = new ListenerProperties(Ordered.HIGHEST_PRECEDENCE);
+    private ListenerProperties chunk = new ListenerProperties(Ordered.HIGHEST_PRECEDENCE);
+    private ListenerProperties skip = new ListenerProperties(Ordered.HIGHEST_PRECEDENCE);
```

Comment:
That depends.  I believe (I'd have to double check) that we run in order on the way in and reverse on the way out (I actually think there is a batch PR to address the consistency of this rule).